python/fi_instrumentation/settings.py

```
@@ -96,3 +99,38 @@ def generate_span_id(self) -> int:
+        }
+
+        response = requests.post(
+            url,
+            headers=headers,
+            json={"eval_template_name": eval_name},
+        )
+
+        response.raise_for_status()
+        return response.json().get("result", {})
+    except Exception as e:
+        raise ValueError(f"Failed to check custom eval template: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)